knative-operator/pkg/controller/knativekafka/knativekafka_controller.go

```diff
@@ -38,6 +40,10 @@ var (
 	role              = mf.Any(mf.ByKind("ClusterRole"), mf.ByKind("Role"))
 	rolebinding       = mf.Any(mf.ByKind("ClusterRoleBinding"), mf.ByKind("RoleBinding"))
 	roleOrRoleBinding = mf.Any(role, rolebinding)
+	// Protects from processing order, if true we should install telemetry
+	// if it is false we need to uninstall in the next delete stage.
+	// We start by assuming no telemetry is available.
+	shouldInstallTelemetry = true
```

Comment:
This should move inside the telemetry implementation.

Reply:
It could be but it is also applied to the current reconciler.